### PR TITLE
Chapter 8

### PR DESCRIPTION
**Pull request by tenyiting**

### CODE REVIEW

**Comment by tenyiting** [comment]
ready

File: app/assets/javascripts/sessions.coffee [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# Place all the behaviors and hooks related to the matching controller here.
```
Comment (ngocvt-0484) [inline app/assets/javascripts/sessions.coffee]: delete this

File: app/assets/stylesheets/sessions.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Sessions controller here.
+// They will automatically be included in application.css.
```
Comment (ngocvt-0484) [inline app/assets/stylesheets/sessions.scss]: samedelete

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,2 +1,15 @@
 class ApplicationController < ActionController::Base
+    protect_from_forgery with: :exception
+    before_action :set_locale
+    include SessionsHelper
+
+    def set_locale
+        I18n.locale = params[:locale] || I18n.default_locale
+        #cureent_user.locale
```
Comment (ngocvt-0484) [inline app/controllers/application_controller.rb]: remove

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -1,2 +1,15 @@
 class ApplicationController < ActionController::Base
+    protect_from_forgery with: :exception
+    before_action :set_locale
+    include SessionsHelper
+
+    def set_locale
+        I18n.locale = params[:locale] || I18n.default_locale
```
Comment (ngocvt-0484) [inline app/controllers/application_controller.rb]: wrong indent

File: app/controllers/application_controller.rb [hunk 5]
```diff
@@ -1,2 +1,15 @@
...
+        #cureent_user.locale
+        #request.env["HTTP_ACCEPT_LANGUAGE"]
+    end
+
+    def default_url_options(options = {})
+        { locale: I18n.locale }.merge options
```
Comment (ngocvt-0484) [inline app/controllers/application_controller.rb]: wrong indent
tab 2 spaces please

File: test/controllers/sessions_controller_test.rb [hunk 6]
```diff
@@ -0,0 +1,9 @@
+require 'test_helper'
+
+class SessionsControllerTest < ActionDispatch::IntegrationTest
```
Comment (ngocvt-0484) [inline test/controllers/sessions_controller_test.rb]: delete this file

**Comment by tenyiting** [comment]
ready

**Review by ghost — COMMENTED** [review]
You have 2 commit, please combine into 1

**Comment by tenyiting** [comment]
ready

**Review by ghost — COMMENTED** [review]
there are some comment not fix yet

**Comment by tenyiting** [comment]
ready

File: app/views/layouts/application.html.erb [hunk 7]
```diff
@@ -6,7 +6,7 @@
     <%= csrf_meta_tags %>
     <%= csp_meta_tag %>
     <%= stylesheet_link_tag "application", media: 'all', "data-turbolinks-track": "reload" %>
-    <%= javascript_pack_tag "application", "data-turbolinks-track": "reload" %>
```
Comment (ngocvt-0484) [inline app/views/layouts/application.html.erb]: why you delete it
please keep it

File: app/views/sessions/new.html.erb [hunk 8]
```diff
@@ -0,0 +1,14 @@
+<% provide :title, t("log_in") %>
+<h1><%= t("log_in") %></h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+      <%= form_for :session, url: login_path do |f| %>
+      <%= f.label :email %>
```
Comment (ngocvt-0484) [inline app/views/sessions/new.html.erb:6]: wrong indent

**Comment by tenyiting** [comment]
ready

**Review by ghost — COMMENTED** [review]
merged

**Pull request merged**